src/sinks/pulsar.rs

```diff
@@ -78,6 +80,8 @@ pub struct PulsarSinkConfig {
     pub acknowledgements: AcknowledgementsConfig,
 
     /// Log field to use as Pulsar message key.
+    #[configurable(metadata(docs::examples = "message"))]
+    #[configurable(metadata(docs::examples = "my_field"))]
     partition_key_field: Option<String>,
```

Comment:
just noticing the generate config function sets this to "message".
But it looks like internally it truly is optional. Just thought it odd there is a default in the generate config.

Reply:
Yeah, but I think that's just used to test the toml serialization/deserialization?

Reply:
I think it's also used to generate default configs with the command "vector generate <...>"

Reply:
Updated this. Thanks.